### ISSUE TITLE
False positive runtime-import-in-type-checking-block (TCH004)

### ISSUE THREAD

**Issue by jschippergoauto** [astral-sh/ruff#9053]
Ruff (0.1.7) gives me a false positive TCH004.

Code:
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from sqlalchemy.orm import Query


def x(q: Query) -> None:
    pass
```

Output of `ruff --isolated --select TCH test.py`:
```
test.py:4:32: TCH004 Move import `sqlalchemy.orm.Query` out of type-checking block. Import is used for more than type hinting.
Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

**Comment by jschippergoauto** [comment]
@charliermarsh Nice, I did not know that! I thought they were not needed at runtime at all. Ruff could be advertised not just as a linter/formatter but as an educational tool as well :laughing:. Thanks!

**Comment by charliermarsh** [comment]
Haha, I learned this too through implementing the rule :joy: I was surprised!

There are a few exceptions... for example, annotated assignments outside of the top-level, e.g. here the annotated isn't required to be available at runtime:

```python
def func():
    x: Query = None
```

(Ruff handles these correctly, or at least it should.)

**Comment by jschippergoauto** [comment]
> There are a few exceptions... for example, annotated assignments outside of the top-level, e.g. here the annotated isn't required to be available at runtime:
> 
> ```python
> def func():
>     x: Query = None
> ```
> 
> (Ruff handles these correctly, or at least it should.)

Double checked and yes it works properly there as well :slightly_smiling_face:. Having worked mostly on statically typed languages I'm surprised the way Python works with this. Thanks for clearing up things.
